### PR TITLE
FAI-7404: lock down user reports coln

### PR DESCRIPTION
**Pull request by ted-faros**
Added a check to prevent regular users from archiving or renaming the `User reports` collections.

### CODE REVIEW

File: src/metabase/api/collection.clj [hunk 1]
```diff
@@ -904,6 +904,23 @@
                             {:card-ids (db/select-ids Card :collection_id (u/the-id collection-before-update))}))]
       (api.card/delete-alert-and-notify-archived! alerts))))
 
+(defn- is-pipeline-reports-coln
+  "Check if the elements of collection's location match a pipeline reports collection."
+  [collection]
+  (let [parents (some->> (str/split (:location collection) #"/")
+                         rest
+                         not-empty
+                         (map #(:name (db/select-one Collection :id (Integer/parseInt %)))))]
```
Comment (eskrm) [inline src/metabase/api/collection.clj:913]: Can we use the existing ancestor functions and map the names instead?

File: src/metabase/api/collection.clj [hunk 2]
```diff
@@ -904,6 +904,23 @@
                             {:card-ids (db/select-ids Card :collection_id (u/the-id collection-before-update))}))]
       (api.card/delete-alert-and-notify-archived! alerts))))
 
+(defn- is-pipeline-reports-coln
```
Comment (eskrm) [inline src/metabase/api/collection.clj:907]: nit: either pass the parent collection in or call this`is-inside-pipeline-reports-coln`?

**Review by eskrm — APPROVED** [review]
LGTM!

File: src/metabase/api/collection.clj (hunk 1 above)
Comment (ted-faros) [reply src/metabase/api/collection.clj:913]: I started with that approach didn't find a suitable function to use.  Do you have something specific in mind?

Comment (ted-faros) [reply src/metabase/api/collection.clj:913]: This was my starting point: https://github.com/faros-ai/metabase/blob/tt/user-rpts/enterprise/backend/src/metabase_enterprise/serialization/names.clj#L86

Comment (ted-faros) [reply src/metabase/api/collection.clj:913]: This is the only method in the code base that parses `:location`

Comment (eskrm) [reply src/metabase/api/collection.clj:913]: I was looking at existing utils in the collection model file: https://github.com/faros-ai/metabase/blob/aion-v46.8/src/metabase/models/collection.clj. Specifically `location-path` and `effective-ancestors`.

File: src/metabase/api/collection.clj [hunk 3]
```diff
@@ -904,6 +904,21 @@
                             {:card-ids (db/select-ids Card :collection_id (u/the-id collection-before-update))}))]
       (api.card/delete-alert-and-notify-archived! alerts))))
 
+(defn- is-inside-pipeline-reports-coln
+  "Check if the elements of collection's location match a pipeline reports collection."
+  [collection]
+  (let [parents (some->> (collection/location-path->ids (:location collection))
```
Comment (ted-faros) [inline src/metabase/api/collection.clj:910]: @eskrm yeah, this is better

File: src/metabase/api/collection.clj (hunk 1 above)
Comment (ted-faros) [reply src/metabase/api/collection.clj]: @eskrm using `location-path->ids`, now.  Better!

**Pull request merged**